From 4bae8e6d95e92b9af998d7f4ced5b388c8637f90 Mon Sep 17 00:00:00 2001
From: Junhao He <hejunhao3@huawei.com>
Date: Mon, 12 Dec 2022 20:27:54 +0800
Subject: [PATCH] coresight: Return the pointer of @pdata when not "fwnode"
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 4bae8e6d95e92b9af998d7f4ced5b388c8637f90
Modified-by-SEL: No


driver inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I5YCYK

--------------------------------------------------------------------------

TRBE device can use the id_tabel to match driver. So if the CoreSight
component does not use the device tree or ACPI to report devices
(@fwnode is NULL), we need to return the pointer to the platform
data @pdata.

Signed-off-by: Junhao He <hejunhao3@huawei.com>
Reviewed-by: Xiongfeng Wang <wangxiongfeng2@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/hwtracing/coresight/coresight-platform.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/hwtracing/coresight/coresight-platform.c b/drivers/hwtracing/coresight/coresight-platform.c
index c594f45319fc..82b900076dbd 100644
--- a/drivers/hwtracing/coresight/coresight-platform.c
+++ b/drivers/hwtracing/coresight/coresight-platform.c
@@ -844,15 +844,15 @@ coresight_get_platform_data(struct device *dev)
 	struct coresight_platform_data *pdata = NULL;
 	struct fwnode_handle *fwnode = dev_fwnode(dev);
 
-	if (IS_ERR_OR_NULL(fwnode))
-		goto error;
-
 	pdata = devm_kzalloc(dev, sizeof(*pdata), GFP_KERNEL);
 	if (!pdata) {
 		ret = -ENOMEM;
 		goto error;
 	}
 
+	if (IS_ERR_OR_NULL(fwnode))
+		return pdata;
+
 	if (is_of_node(fwnode))
 		ret = of_get_coresight_platform_data(dev, pdata);
 	else if (is_acpi_device_node(fwnode))
-- 
2.33.0

